src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -548,4 +548,13 @@ public List<Dataset> findDatasetsByIds(List<Integer> datasetIds) {
     return datasetDAO.findDatasetsByIdList(datasetIds);
   }
 
+  public Study addDatasetsToStudy(Set<Integer> datasetIds, Study study) {
+    List<Integer> datasetIdList = new ArrayList<>(datasetIds);
+    List<Dataset> datasets = findDatasetsByIds(datasetIdList);
+    for (Dataset dataset : datasets) {
+      study.addDataset(dataset);
+    }
+    return study;
+  }
```

Comment:
I think maybe my previous feedback was not clear. The goal of the service class is to populate the object requested, but not necessarily split that up into different methods. The resource class should ask the service class to get the study once, along with its datasets, so that we don't need to do population logic inside the resource class. If I were refactoring this, I'd have one method here in the service class, `getStudyWithDatasetsById(Integer studyId)`:
```
  public Study getStudyWithDatasetsById(Integer studyId) {
    Study study = studyDAO.findStudyById(studyId);
    if (Objects.isNull(study)) {
      throw new NotFoundException("Study not found");
    }
    if (Objects.nonNull(study.getDatasetIds()) && !study.getDatasetIds().isEmpty()) {
      List<Dataset> datasets = findDatasetsByIds(new ArrayList<>(study.getDatasetIds()));
      study.addDatasets(datasets);
    }
    return study;
  }
```
I would also add a new method to `Study` to add all datasets at once instead of looping and adding them individually.